### ISSUE TITLE
gnome-shell, xorg, i915: crash when adjusting screen arguments at high-resolution

### ISSUE THREAD

**Issue by lmy441900** [AOSC-Dev/aosc-os-abbs#404]
Just now I turned my screen 90 degrees, and when I was turning it back, GNOME, X.Org, and i915 are all crashed.

Log is attached. Several attempts to switch display arguments (all failed with core dumps of `gnome-shell` and `i915`, where `gnome-shell` claims that it received an error from Xorg) can be seen.

[journal.txt](https://github.com/AOSC-Dev/aosc-os-abbs/files/494979/journal.txt)

**Comment by lmy441900** [comment]
BTW, all settings remain untouched, and thus my screen cannot be rotated back.

**Comment by lmy441900** [comment]
Some other information:

- GNOME 3.20.2
- After rotating the screen, there is a **looooong** torn-like vertical line on the center line. Can be seen when moving windows.
  - It even looks like a **lightening** in the middle of the screen. OMG
- Before the failure, I tried to switch the refreshing rate from "60 Hz" to "60 Hz (NTSC)". After switching the "60 Hz" option disappeared.
- Adjusting the refreshing rate, like to "50 Hz", is successful, while rotating the screen causes the big crash.

**Comment by MingcongBai** [comment]
Which connection did you use for your external screen?

**Comment by lmy441900** [comment]
@MingcongBai HDMI

**Comment by MingcongBai** [comment]
Okay.

**Comment by lmy441900** [comment]
@MingcongBai 啊

**Comment by MingcongBai** [comment]
@lmy441900 Tested that the crash could not be reproduced with `xf86-video-modesetting` (included in `xorg-server`). However, I did find a set of patches, included with Fedora 24's kernel, regarding to the errors seen in the journal provided [here](https://github.com/AOSC-Dev/aosc-os-abbs/files/494979/journal.txt):

http://pkgs.fedoraproject.org/cgit/rpms/kernel.git/tree/kernel.spec?h=f24#n603

This might help, I will include these as a part of the 4.7.5 update.

**Comment by MingcongBai** [comment]
Just as a suggestion, you could always try to provide info about the hardware you are using (if not for I knowing your Dell has a Skylake board...).

**Comment by lmy441900** [comment]
> you could always try to provide info about the hardware you are using

Fine.

**Comment by MingcongBai** [comment]
Is this issue resolved with the latest updates to `mutter`, `gnome-shell`, and `linux-kernel-4.7.5`? If yes, please close this.

**Comment by lmy441900** [comment]
> Is this issue resolved with the latest updates to mutter, gnome-shell, and linux-kernel-4.7.5? If yes, please close this.

A test will be performed on Oct. 3.

**Comment by lmy441900** [comment]
Just now I had some tests:

- Using the updated kernel 4.7.5-aosc-main to perform an 2nd screen's rotation.
  - Tearing still appears;
  - Switching 2nd screen's direction from vertical to horizontal still causes a crash.

Some more tests:

- Lower the 2nd screen's resolution (say, from 1920x1080 to 1024x768). This operation seems to be safe;
- With a lower resolution (initially 1920x1080), rotate the screen. **No error occurred.**

In conclusion:

- **At a high resolution X Window just cannot handle the 2nd screen's rotation.**

**Comment by MingcongBai** [comment]
GNOME 3.22 is now released, could you still reproduce this issue?

**Comment by bobcao3** [comment]
> At a high resolution X Window just cannot handle the 2nd screen's rotation.
It could be X Window system is having hardware stress when you rotate the second screen. As there are more pixels on that screen, the GPU or CPU need to calculate more pixels.

**Comment by MingcongBai** [comment]
No response. Closing.
